src/runtime/database.cpp

```diff
@@ -1091,6 +1126,20 @@ static std::vector<JobReflection> find_all(const Database *db, sqlite3_stmt *que
   return out;
 }
 
+std::vector<std::string> Database::get_outputs() const {
```

Comment:
Maybe this should have been get_and_remove_outputs? So that the remove_output_files and get_output_files could be in the same begin/end_txn